e2e/perf/download-artifacts.py

```
@@ -0,0 +1,177 @@
+    sys.exit(1)
+
+
+def list_artifacts(owner: str, repo: str, run_id: int, token: str) -> list[dict]:
+    """List all artifacts for a workflow run."""
+    url = f"https://api.github.com/repos/{owner}/{repo}/actions/runs/{run_id}/artifacts"
+    headers = {
+        'Authorization': f'token {token}',
+        'Accept': 'application/vnd.github.v3+json'
+    }
+
+    response = requests.get(url, headers=headers)
```

Ruff (0.16.0)

[error] Probable use of `requests` call without timeout

(S113)